### PR TITLE
Fix CohereASR training loss double shift

### LINKED ISSUE #46894: CohereASR training-loss double-shift bug (same pattern as Moonshine #46784)
### System Info

- `transformers` version: 5.13.0.dev0 (main branch at commit dc5a497c18)
- - Platform: Windows
- - - Python version: 3.x
- - - - Using GPU: N/A (code-level bug, not runtime)

### Who can help?

@eustlb @ebezzam @vasqu

### Information

- [x] The official example scripts
- [ ] My own modified scripts

### Tasks

- [ ] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [ ] My own task or dataset (give details below)

### Reproduction

aIn `src/transformers/models/cohere_asr/modular_cohere_asr.py`, `CohereAsrForConditionalGeneration.forward()` has a double-shift bug:

1. Labels are shifted right via `shift_tokens_right()` (line 483) to create `decoder_input_ids`
2. 2. Then `self.loss_function()` is called (line 503), which maps to `ForCausalLMLoss`
3. 3. `ForCausalLMLoss` shifts labels again internally (`labels[..., 1:]` in `loss/loss_utils.py` line 63-64)
This means the model trains against `labels[..., 1:]` instead of `labels`.

This is the **exact same bug** that was fixed in Moonshine in PR #46784 (commit d8c235494e). The Moonshine fix replaced `self.loss_function()` with `CrossEntropyLoss()`, but `CohereAsr` — which inherits from Moonshine and overrides `forward` — was not updated.

```python
# Buggy code (line 503 in modular_cohere_asr.py):
loss = self.loss_function(logits=logits, labels=labels, vocab_size=self.config.vocab_size)

# Should be:
loss_fct = CrossEntropyLoss()
loss = loss_fct(logits.reshape(-1, self.config.vocab_size), labels.reshape(-1))
```

### Expected behavior

Training loss should be computed directly against `labels` without any additional shifting, since `shift_tokens_right()` already handles the decoder input alignment. This matches the behavior of Whisper, Bart, and the now-fixed Moonshine model. The fix should use `CrossEntropyLoss()` instead of `self.loss_function()`, as done in Moonshine PR #46784.

### PR DESCRIPTION
**Pull request by w-Jessamine**
[CI](https://transformers-ci.lor-e.huggingface.cool/d/pytest-observability-by-pr/pytest-observability-branch?var-pr=46958)

## Summary

Fixes #46894.
Supersedes #46947, which was closed after the fork branch was renamed.

CohereASR right-shifts `labels` into `decoder_input_ids`, then used `self.loss_function` (`ForCausalLMLoss`) to compute the training loss. That loss shifts labels internally again, so training targets were effectively `labels[..., 1:]`.

This switches CohereASR to a plain `CrossEntropyLoss`, matching the Moonshine fix and seq2seq models that already align decoder inputs from labels before computing loss.

## Tests

- `PYTHONPATH=src /tmp/transformers-pr-venv/bin/python -m pytest tests/models/cohere_asr/test_modeling_cohere_asr.py::CohereAsrModelTest::test_training_loss_no_double_shift -q`

---

### CI recap

**Dashboard:** [View test results in Grafana](https://transformers-ci.lor-e.huggingface.cool/d/pytest-observability-by-pr/pytest-observability-branch?var-pr=46958)
**Latest run:** [28370058252](https://github.com/huggingface/transformers/actions/runs/28370058252)
**Result:** `success` | Grafana metrics are not available yet.

### CODE REVIEW

**Review by eustlb — COMMENTED** [review]
#47090 is the correct fix IMO, see https://github.com/huggingface/transformers/pull/47090#pullrequestreview-4637793592 for explaination
cc @vasqu @SunMarc if you agree
